packages/sync-actions/src/utils/common-actions.js

```diff
@@ -2,6 +2,12 @@ import isNil from 'lodash.isnil'
 import clone from './clone'
 import * as diffpatcher from './diffpatcher'
 
+export function getIsAnEmptyValue(value, shouldOmitEmptyString) {
+  const isEmptyString = v => Boolean(typeof v === 'string' && v.trim() === '')
```

Comment:
As another thread: We could also change the signature so that it receives what it's meant to treat as an empty value. `isAnEmptyValue(value, [undefined, null]` per default and with `shouldOmitEmptyString` it just receives `isAnEmptyValue(value, [undefined, null, '']`. What do you think?

Reply:
hmmm... yes I get the idea, but I wonder what that signature would look like:

On the top of my head, that could be

```js
const isEmptyValue = (value, emptyValues) => emptyValues.includes(value)
```

but we still would need the trimming, since the user could potentially pass in `' '` as a way to trick the update-actions previously.

Reply:
I actually prefer the original approach

Reply:
No, the user can't specify that. The user would not specify the `emptyValues`. A consumer would still only specify `shouldOmitEmptyString` which translates to a static `[undefined, null, '']`.

Reply:
> No, the user can't specify that. 

no, they don't specify the config, but they do specify the value? that is what I was referring to with _since the user could potentially pass in ' ' as a way to trick the update-actions previously._

Reply:
Sorry for still failing to get my point across. We don't change the user facing configuration options

1. We keep the config as is with `shouldOmitEmptyString`
2. We however don't pass around the `shouldOmitEmptyString` so deeply instead the `getIsAnEmptyValue` becomes

```js
const normalizeValue = value => typeof value === 'string' ? value.trim() : value;
const createIsEmptyValue = emptyValues => value => 
  emptyValues.some(emptyValue => emptyValue === normalizeValue(value))
```

In this line https://github.com/commercetools/nodejs/pull/592/files#diff-d544c8548498969be35940fcda6c3386R36

We can then there do a

```js
const defaultEmptyValues = [null, undefined];
const isEmptyValue = createIsEmptyValue(options.shouldOmitEmptyString ? emptyValues.concat('') : emptyValues)
```

To be that would be easier to understand which is subjective obviously. Again as previously stated I am fine either way. So feel free to pick or drop.

Reply:
> We however don't pass around the shouldOmitEmptyString so deeply instead the getIsAnEmptyValue becomes

okay, now I get it 😄 
I like your suggested approach better. update coming up!

Reply:
df7aa27dd06bc5cb9ce95cfb03976e3975f809ce